pkg/spiffe/spiffe_test.go

```diff
@@ -661,3 +663,76 @@ func TestGetGeneralCertPoolAndVerifyPeerCert(t *testing.T) {
 		})
 	}
 }
+
+func TestExpandWithTrustDomains(t *testing.T) {
```

Comment:
The function has all sorts of tricky string parsing stuff. if we are going to keep that, can we add cases for bad inputs? we need to make sure it won't panic on `spife//ns/ns/ns/ns/ns` or other invalid inputs

Reply:
Indeed. though as it's focusing on the prefix part, (we have some variants on the suffix for the annotation based like vm or ns/sa pattern)

I added a invalid for the invalid prefix and truts domain case.

Reply:
resolving, feel free to reopen.